src/www/ui/core-auth.php

```diff
@@ -194,53 +194,109 @@ function getIP()
     return (@$_SERVER['REMOTE_ADDR']);
   }
 
-  /**
-   * \brief This is only called when the user logs out.
-   */
   public function Output()
   {
     $userName = GetParm("username", PARM_TEXT);
     $password = GetParm("password", PARM_TEXT);
     $referrer = GetParm("HTTP_REFERER", PARM_TEXT);
-    if (empty($referrer))
+    $openidProviderLogin = GetParm("openidProviderLogin", PARM_TEXT);
+
+    global $SysConf;
+    $openIdConnect = $SysConf["OPENID_CONNECT"];
+    $provider_name = $openIdConnect["provider_name"];
+    $provider_url = $openIdConnect["provider_url"];
+    $client_id = $openIdConnect["client_id"];
+    $client_secret = $openIdConnect["client_secret"];
+    $redirect_url = $openIdConnect["redirect_url"];
+    $claim = $openIdConnect["claim"];
+
+    $isOpenIDLogin = !empty($_GET["code"]) || !empty($openidProviderLogin);
+    if($isOpenIDLogin)
     {
-      $referrer = GetArrayVal('HTTP_REFERER', $_SERVER);
+
+      $client = new OpenIDConnectClient($provider_url, $client_id, $client_secret);
```

Comment:
there is no import corresponding to the new dependency? Please add an explicit import (if applicable).